target/linux/qualcommax/image/ipq807x.mk

```diff
@@ -91,6 +91,38 @@ define Device/edimax_cax1800
 endef
 TARGET_DEVICES += edimax_cax1800
 
+define Device/linksys_mx4200v1
+	$(call Device/FitImage)
+	$(call Device/UbiFit)
+	DEVICE_VENDOR := Linksys
+	DEVICE_MODEL := MX4200v1
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+        KERNEL_SIZE := 6144k
+	SOC := ipq8174
+	UBINIZE_OPTS := -E 5    # EOD marks to "hide" factory sig at EOF
+	IMAGES += factory.bin
+	IMAGE/factory.bin  := append-kernel | pad-to $$$$(KERNEL_SIZE) | append-ubi | linksys-image type=MX4200
+	DEVICE_PACKAGES := kmod-leds-pca963x ipq-wifi-linksys_mx4200 kmod-ath10k-ct-smallbuffers kmod-usb-storage-uas kmod-usb-storage-extras kmod-usb-dwc2-pci kmod-usb-uhci kmod-fs-ext4 
+endef
+TARGET_DEVICES += linksys_mx4200v1
+
+define Device/linksys_mx4200v2
+	$(call Device/FitImage)
+	$(call Device/UbiFit)
+	DEVICE_VENDOR := Linksys
+	DEVICE_MODEL := MX4200v2
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+        KERNEL_SIZE := 6144k
+	SOC := ipq8174
+	UBINIZE_OPTS := -E 5    # EOD marks to "hide" factory sig at EOF
+	IMAGES += factory.bin
+	IMAGE/factory.bin  := append-kernel | pad-to $$$$(KERNEL_SIZE) | append-ubi | linksys-image type=MX4200
+	DEVICE_PACKAGES := kmod-leds-pca963x ipq-wifi-linksys_mx4200 kmod-usb-storage-uas kmod-usb-storage-extras kmod-usb-dwc2-pci kmod-usb-uhci kmod-fs-ext
```

Comment:
Remove USB and EXT4 kmods, those are option for the device to function and can be easily installed.